provider/datagen/src/transform/cldr/time_zones/convert.rs

```diff
@@ -119,45 +119,48 @@ impl Location {
-                                        Some(bcp47) => place
-                                            .exemplar_city()
-                                            .map(|city| vec![(bcp47, city)])
-                                            .unwrap_or_default(),
-                                        None => panic!("Cannot find bcp47 for {key:?}."),
+                .filter_map(|(bcp47, bcp47_tzid_data)| {
+                    bcp47_tzid_data
+                        .alias
+                        .as_ref()
+                        .map(|aliases| (bcp47, aliases))
+                })
+                .filter_map(|(bcp47, aliases)| {
+                    aliases
+                        .split(' ')
+                        .find_map(|alias| {
+                            let mut alias = alias.split('/');
+                            Some((
+                                bcp47,
+                                match other
+                                    .time_zone_names_resource
+                                    .zone
+                                    .0
+                                    .get(alias.next()?)?
+                                    .0
+                                    .get(alias.next()?)?
+                                {
+                                    LocationOrSubRegion::Location(place) => place,
+                                    LocationOrSubRegion::SubRegion(region) => {
+                                        region.get(alias.next()?)?
                                     }
                                 }
-                                LocationOrSubRegion::SubRegion(region) => region
-                                    .iter()
-                                    .filter_map(|(inner_key, place)| {
-                                        let mut key = key.clone();
-                                        key.push('/');
-                                        key.push_str(inner_key);
-                                        match bcp47_tzid_data.get(&key) {
-                                            Some(bcp47) => {
-                                                place.exemplar_city().map(|city| (bcp47, city))
-                                            }
-                                            None => panic!("Cannot find bcp47 for {key:?}."),
-                                        }
-                                    })
-                                    .collect::<Vec<_>>(),
+                                .exemplar_city()?,
+                            ))
+                        })
+                        .or_else(|| {
+                            let alias = aliases.split(' ').next().unwrap();
+                            if alias.contains('/') && !alias.starts_with("Etc") {
+                                Some((
+                                    bcp47,
+                                    alias.split('/').next_back().unwrap().replace('_', " "),
```

Comment:
I think you're getting the literal "Montreal" strings from this fallback.

Note that America/Monteal says in data that it is deprecated

```
        "camtr": {
          "_deprecated": true,
          "_description": "Montreal, Canada",
          "_alias": "America/Montreal"
        },
```

I think we should do one of the following

1. If we can't find translations, skip it rather than adding it with this fallback, or
2. Don't include translations for any time zones that are marked as deprecated

Reply:
https://github.com/unicode-org/cldr/pull/2796 Fixes this

Reply:
Actually we need to pre-43 CLDR, so I've added a special case for `camtr`.